src/main/java/net/minecraftforge/client/event/InputEvent.java

```diff
@@ -199,4 +201,72 @@ public int getModifiers()
             return this.modifiers;
         }
     }
+
+    /**
+     * This event fires when one of the keybindings that by default involves clicking the mouse buttons
+     * is triggered.
+     *
+     * These key bindings are use item, pick block and attack keybindings. (right, middle and left mouse click)
+     * In the case that these key bindings are re-bound to a keyboard key the event will still be fired as normal.
+     */
+    @Cancelable
+    public static class ClickInputEvent extends InputEvent
+    {
+        private final int button;
+        private final KeyBinding keyBinding;
+        private final Hand hand;
+        private boolean handSwing = true;
+        public ClickInputEvent(int button, KeyBinding keyBinding, Hand hand)
+        {
+            this.button = button;
+            this.keyBinding = keyBinding;
+            this.hand = hand;
+        }
+
+        /**
+         * Set to false to disable the hand swing animation.
+         * @throws UnsupportedOperationException If this is not an attack input.
```

Comment:
This is incorrect. The swinging action does happen when you interact or place a block.

Reply:
The right click indeed has a swing animation too, but there wasn't any code yet to specifically block the swing animation attached to the event. I've added it now though.